core/runtime/TRTEngine.h

```diff
@@ -102,7 +103,8 @@ struct TRTEngine : torch::CustomClassHolder {
   std::vector<at::Tensor> input_buffers = {};
   std::vector<at::Tensor> output_buffers = {};
   std::string shape_key;
-
+  bool cudagraphs_enabled = false;
```

Comment:
What if we made this like an enum that describes the runtime mode `STANDARD, SUBGRAPH_CUDAGRAPHS, WHOLE_GRAPH_CUDAGRAPHS` so we avoid the checks in execute engine?

Reply:
Actually why does the engine need to know if its in whole graph cudagraph mode? seems like it doesnt actually change the behavior from "STANDARD" mode